### DISCUSSION TITLE
Add include/exclude devices to Fritzbox Tools

### DISCUSSION THREAD

**Discussion by emufan** [home-assistant/architecture#618] [General]
According to some discussions in the community and a reported issue:

Is it really not an option or not planned to implement an include/exclude settings in the integration?

Yes, default is, that the tracker-entities are disabled (but devices not disabled), BUT in the meantime it is more than hard to have the overview about “real” disabled entities, which I have disabled because of reasons, etc. and the 99% fritzbox tool devices, which I never ever will use as trackers and are disables in the entity list. esp. as these are now 2x with the new internet access switches.

A simple include box, where I can set my 2 from 200 devices (not all available, but I do not press "forget not existing devices every day in fritzbox, because some of them have settings, …), which I want to have as trackers during integration setup or (preferred) always there. Same as in other integrations like alexa media player, …?

Unfortunately, I’m not able to develop this (not a developer), but the 198 or now 396 existing entities without any reason in my entity list … this is really jumping the shark.

The integration generates active devices and non-active tracker and switch entities for 99% of my WLAN and LAN devices, which I will never ever track or switch. And I think that this is for all of the other users as well.

I have hundreds of devices and entities in y HA lists only from this integration and can not see and focus on the relevant other entities anymore.

Instead of heaving e.g. 200 devices and 400 entities I only need currently 2 devices or 2 entities from this bunch of generated ones. 2 from 400. And all the other are fill up my HA setup.

In other integrations, there are options, which entities should the integration generate or which devices should be included and included (2 from 200 in my example). And this is my suggestion to stop filling up the HA systems from this integration.

A simple include text field, where I can enter for which devices the integration setup generates tracker entities.
Another include text field, where I can enter for which devices the integration setup generates internet access switches (0 from 200 here).

**Simple textbox, comma separated values. My proposal (as you are using this already and is unique): mac address,**

If empty, everything as before. Loop over all fritzbox network devices and setup device and the corresponding tracker and switch.

If not empfy, loop over all fritzbox network devices but setup only device and the corresponding tracker and switch, if the device is in the include textbox (see proposal check if max address is in comma separated mac-address list in include text field).

In my opinion, this is not a big change, only the check if in textbox otherwise skip. But would bring a big value and aviods to setup all the unwanted devices and entities.

Esp. with the last updates. Just assume, that someone only set up this integration, because netmonitor is not available stand alone anymore and only want to have the netmonitor entities. And he will get? Hundrets of devices, entities, switches, ... I hope you get the point. At least i do not know any of such multi-ingeration, which generates so much unwanted stuff without having the users choise, esp. here with the unfiltered devices list coming from Fritzbox.

But why is it not possible to check in common.py here

```
    def scan_devices(self, now: datetime | None = None) -> None:
        """Scan for new devices and return a list of found device ids."""
```

not only against already generated devices but also against an include list and skip every not already generated device, if it is not in that include list (if the include list is not empty)?

Same here in device_tracker.py

```
def _async_add_entities(
    router: FritzBoxTools,
    async_add_entities: AddEntitiesCallback,
    data_fritz: FritzData,
) -> None:
    """Add new tracker entities from the router."""
```

check not only again already created trackers, but again the device list (mac address include list) as well and skip adding new for all, which are not in the include list (if the include list is not empty).

And last but least for and in switch.py in def profile_entities_list.

All this is (afaik, but already said, not a programmer, only able to read and copy and paste some lines) only in this integration and the chance to skip generating the 99% unwanted devices and entities as well.

As there are lots of merges/changes in this integration, please think about this.

**Comment by MartinHjelmare** [comment]
We already have this issue where this is discussed:
https://github.com/home-assistant/architecture/issues/307

**Reply by emufan** [reply]
Yes and no. Here this is not only about the tracker entities, but for the generated devices as well and the generated switch entities. This special integration is generating hundreds of devices and entities, even if you only want to have e.g. 2 or perhaps 0, because you only want to have the netmonitor entities.

I created an issue here https://github.com/home-assistant/core/issues/54340 as well, but Simone want to have it discusssed here, because in his point of view it is not an issue but only a feature request to not have hundrets of unwanted entities and devices.

**Reply by chemelli74** [reply]
Yes, I want a architecture approach and not a integration by integration one ;-)

IMHO, we should discuss only about device tracker devices.
Switches and sensors instead should never be filtered but just disabled if user is not interested into those data.

Simone

**Reply by emufan** [reply]
> Switches and sensors instead should never be filtered but just disabled if user is not interested into those data.

Really?

Please consider my example. You have recently merged and deleted the fritz netmonitor integration. The user, who only need and wants the netmonitor has now to enable to tools integration. But instead of having now the netmonitor entities only, he gets now perhaps (depending on his fritzbox) 250 active devices and 500 (!) deactivated switch and sensor entities.

This is messing up HA extremely. Between the deactivated entities, you are not able to see, search and find the relevant entities deactivated by purpose.

Why do you really not see this problem? Completely not understand this. In this arguments, you think a user can setup all integrations, even if not needed and then deactivate later, what he does not want? No. The setup should always contain as less as possible.,

Can you explain **why** you do  not want to give the users, who get messed up with all theses unwanted entities an include-textbox during integration setup and check again this in the given positions? Those (like you), who are not wanting to use this include list, just leave it empty and everything is as before.

As given in my examples for alexa_media_player, google cast, etc.

**Reply by mib1185** [reply]
Honestly I`m with @emufan and had already same discussion in the past - see https://github.com/home-assistant/core/pull/50574#discussion_r632980924

**Reply by Flop2006** [reply]
Is there already something new here?
I actually only need two device trackers, but after I saw that the FritzBox Tools creates over 200 devices, some of which cannot be deleted, I uninstalled the tools again.
It would be nice if there was a more user-friendly way.

**Reply by chemelli74** [reply]
Use cleanup service to delete stale devices. Select and disable all the other.

Simone
